src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -124,6 +126,655 @@ const submitEncryptModeForm: ControllerHandler<
+    payment.paymentIntentId = paymentIntentId
+    payment.pendingSubmissionId = pendingSubmissionId
+    try {
+      await payment.save()
+    } catch (err) {
+      logger.error({
+        message: 'Error updating payment document with payment intent id',
+        meta: {
+          ...logMeta,
+          pendingSubmissionId,
+          paymentIntentId,
+        },
+        error: err,
+      })
+      // Cancel the payment intent if saving the document fails.
+      try {
+        await stripe.paymentIntents.cancel(paymentIntent.id, {
+          stripeAccount: targetAccountId,
+        })
+      } catch (stripeErr) {
+        logger.error({
+          message: 'Failed to cancel Stripe payment intent',
+          meta: {
+            ...logMeta,
+            pendingSubmissionId,
+            paymentIntentId,
+          },
+          error: err,
+        })
+      }
+      // Regardless of whether the cancellation succeeded or failed, block the
+      // submission so that user can try to resubmit
+      return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({
+        message:
+          'There was a problem updating the payment document. Please try again.',
+      })
+    }
+
+    logger.info({
+      message: 'Saved payment document to DB',
+      meta: {
+        ...logMeta,
+        pendingSubmissionId,
+        paymentIntentId,
+        paymentId,
+      },
+    })
+
+    return res.json({
+      message: 'Form submission successful',
+      submissionId: pendingSubmissionId,
+      timestamp: (pendingSubmission.created || new Date()).getTime(),
+      paymentData: { paymentId },
+    })
+  }
+  /**
+   * End of Payment Forms Submission Flow
+   */
+
+  const submission = new EncryptSubmission(submissionContent)
+
+  try {
+    await submission.save()
+  } catch (err) {
+    logger.error({
+      message: 'Encrypt submission save error',
+      meta: {
+        action: 'onEncryptSubmissionFailure',
+        ...createReqMeta(req),
+      },
+      error: err,
+    })
+    return res.status(StatusCodes.BAD_REQUEST).json({
+      message:
+        'Could not send submission. For assistance, please contact the person who asked you to fill in this form.',
+      submissionId: submission._id,
+    })
+  }
+
+  const submissionId = submission.id
+  logger.info({
+    message: 'Saved submission to MongoDB',
+    meta: {
+      ...logMeta,
+      submissionId,
+      formId,
+      responseMetadata,
+    },
+  })
+
+  // TODO 6395 make responseMetadata mandatory
+  if (responseMetadata) {
+    reportSubmissionResponseTime(responseMetadata, {
+      mode: 'encrypt',
+      payment: 'true',
+    })
+  }
+
+  // Send success back to client
+  res.json({
+    message: 'Form submission successful.',
+    submissionId,
+    timestamp: (submission.created || new Date()).getTime(),
+  })
+
+  return await performEncryptPostSubmissionActions(
+    submission,
+    incomingSubmission.responses,
+  )
+}
+
+const submitStorageModeForm: ControllerHandler<
```

Comment:
if you really want to split, should duplicate the unit tests too

Reply:
refactored in [132021e](https://github.com/opengovsg/FormSG/pull/6595/commits/132021ee7a92d805a4482dd8a8ec8073b243ddb9) so this is no longer applicable